cv_bridge/src/cv_bridge.cpp

```diff
@@ -532,12 +533,11 @@ CvImageConstPtr cvtColorForDisplay(const CvImageConstPtr& source,
       // Let's decide upon an output format
       if (enc::numChannels(source->encoding) == 1)
       {
-        if (source->encoding == enc::TYPE_32SC1)
+        if ((source->encoding == enc::TYPE_32SC1) ||
```

Comment:
When colormap==-1 (this is default for compatibility with previous behavior), the encoding was force be changed to grayscale. (the encoding was ignored), so maybe this does not change the behavior so much.
But if you are concerning about the ambiguousness of encoding, I can change it. (but I'm not sure what encoding is correct when encoding was not specified as argument for cvtColorForDisplay...)